crates/autopilot/src/run_loop.rs

```diff
@@ -318,59 +309,102 @@ impl RunLoop {
             )
             .await;
 
-        let futures = self
-            .drivers
-            .iter()
-            .enumerate()
-            .map(|(index, driver)| async move {
-                let result =
-                    match tokio::time::timeout(SOLVE_TIME_LIMIT, driver.solve(request)).await {
-                        Ok(inner) => inner,
-                        Err(_) => Err(anyhow!("timeout")),
-                    };
-                (index, result)
-            })
-            .collect::<Vec<_>>();
-        let results = futures::future::join_all(futures).await;
-        results
-            .into_iter()
-            .filter_map(|(index, result)| match result {
-                Ok(result) => Some((index, result)),
+        futures::future::join_all(self.drivers.iter().map(|driver| async move {
+            let start = Instant::now();
+            (start, self.solve(driver, request).await)
+        }))
+        .await
+        .into_iter()
+        .zip(&self.drivers)
+        .fold(Vec::new(), |mut solutions, ((start, result), driver)| {
+            for solution in match result {
+                Ok(solutions) => {
+                    Metrics::get().solve_ok(driver, start);
```

Comment:
This metric doesn't seem useful since it will always report the exact same solving time for all solvers because you use `join_all` which will only return when the last solvers finishes.
If you want to fetch individual timings you can collect the futures into `FuturesUnordered` which behaves like a stream and yields the results exactly when they become available.

Alternatively you could move most of the logic currently inside `.fold()` into the future prepared in `join_all()`. That would be the easiest to read, gives accurate timings and can continue to use `join_all()`.

Reply:
Oops!

Reply:
I changed it to have the futures do `start.elapsed` instead of passing in the `Instant`.

Reply:
d84b7dc3d95bb7aeb93b7940106a6174e71d69b6